.github/workflows/action.yml

```diff
@@ -0,0 +1,41 @@
+name: Specs
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+  pull_request_target:
+
+jobs:
+  test:
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        ruby-version:
+          - "2.4"
+          - "2.5"
+          - "2.6"
+          - "2.7"
+          - "3.0"
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby-version }}
+          bundler-cache: true
+      - name: Test & publish code coverage
+        uses: paambaati/codeclimate-action@v2.7.4
```

Comment:
Is there an advantage of using this action, rather than simply using `curl` to download the binary like the TravisCI script does?

Reply:
I didn't read every line of the action super carefully (and I am not familiar with the cc test reporter), but I think it mainly sets up [required environment varibales](https://github.com/paambaati/codeclimate-action/blob/0f8af43fca84b500025ca48b581bcff933244252/src/main.ts#L39-L53) and handles some edge cases.

Reply:
I'd prefer to use the `curl` approach as that's the official route, rather than depending on a third-party action which may need to be reviewed manually or bumped via dependabot. If you **do** want to use it, please use the commit hash of latest release (`v2.7.5`) -- that's a known good point :slightly_smiling_face: .

Reply:
I locked the version of the action to the commit of the latest release, this should make it a bit safer and still use the same code even if the version is re-tagged. It looks like the action is semi-official, codeclimate reference it in their documentation: https://docs.codeclimate.com/docs/github-actions-test-coverage